### ISSUE TITLE
ePub reading interface not working?

### ISSUE THREAD

**Issue by edoreld** [gustavklopp/LingL#34]
How am I supposed to read and annotate words from an ePub?

I uploaded my Lord of the Rings epub in german. It shows 0 words to do, there are no words in blue, and when I make a large selection it unhelpfully asks whether I want to ""LingLibrify" this text of the webpage?", which does nothing when I say yes.

**Comment by chengxicn** [comment]
I've encountered same issue.

**Comment by gustavklopp** [comment]
I didn't test with an ePUB yet. Can you upload or send the link of the ePUB that you've tried please?

**Comment by edoreld** [comment]
Yes, Github doesn't support ePub so here's a link to a file host I trust: https://www.mediafire.com/file/mhe31xovxegjbm4/Der+Herr+der+Ringe+-+Tolkien,+John+R.epub/file

**Comment by gustavklopp** [comment]
Ok, I've made some modifications to improve this.
You can try today's release.
It's recommended to process at max one page after another: You can try bigger but you'll have to wait more.
In fact, with an EPUB, It's always slower than processing a raw text: The explanation is that we need to keep the layout of the EPUB: Behind an EPUB, it's in fact an assembly of numerous bits of texts. It means that we need to process the text small bit by small bit, and than put them back at the place they were. Instead of processing it in one time with a raw text.
